### PR TITLE
Adds a signal for storage exit and insert, uses it to make trading cards zoom properly

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Adds a signal to the base of on_exit/enter_storage(), uses it to make trading cards zoom in properly

## Why It's Good For The Game

I want my cards fullscreen in the backpack damnit 

## Changelog
:cl:
add: Trading cards will now ZOOM AND ENHANCE in your bag properly.
/:cl:

### CODE REVIEW

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -741,10 +741,12 @@
 
 /// called when this atom is removed from a storage item, which is passed on as S. The loc variable is already set to the new destination before this is called.
 /atom/movable/proc/on_exit_storage(datum/component/storage/concrete/S)
+	SEND_SIGNAL(src, CONSIG_STORAGE_EXITED)
 	return
```
Comment (Rohesie) [inline code/game/atoms_movable.dm:745]:
```suggestion

```

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -741,10 +741,12 @@
...
 
 /// called when this atom is added into a storage item, which is passed on as S. The loc variable is already set to the storage item.
 /atom/movable/proc/on_enter_storage(datum/component/storage/concrete/S)
+	SEND_SIGNAL(src, COMISG_STORAGE_ENTERED)
 	return
```
Comment (Rohesie) [inline code/game/atoms_movable.dm:750]:
```suggestion

```

File: code/game/atoms_movable.dm [hunk 3]
```diff
@@ -741,10 +741,12 @@
 
 /// called when this atom is removed from a storage item, which is passed on as S. The loc variable is already set to the new destination before this is called.
 /atom/movable/proc/on_exit_storage(datum/component/storage/concrete/S)
+	SEND_SIGNAL(src, CONSIG_STORAGE_EXITED)
```
Comment (Rohesie) [inline code/game/atoms_movable.dm:744]:
```suggestion
/atom/movable/proc/on_exit_storage(datum/component/storage/concrete/master_storage)
	SEND_SIGNAL(src, CONSIG_STORAGE_EXITED, master_storage)
```

File: code/game/atoms_movable.dm [hunk 4]
```diff
@@ -741,10 +741,12 @@
...
 	return
 
 /// called when this atom is added into a storage item, which is passed on as S. The loc variable is already set to the storage item.
 /atom/movable/proc/on_enter_storage(datum/component/storage/concrete/S)
+	SEND_SIGNAL(src, COMISG_STORAGE_ENTERED)
```
Comment (Rohesie) [inline code/game/atoms_movable.dm:749]:
```suggestion
/atom/movable/proc/on_enter_storage(datum/component/storage/concrete/master_storage)
	SEND_SIGNAL(src, COMISG_STORAGE_ENTERED, master_storage)
```

File: code/game/objects/items/tcg/tcg.dm [hunk 5]
```diff
@@ -130,10 +140,16 @@ GLOBAL_LIST_EMPTY(cached_cards)
...
 
 /obj/item/cardpack/dropped(mob/user, silent)
 	. = ..()
+	zoom_out()
+
+/obj/item/cardpack/proc/zoom_in()
+	transform = matrix()
+
+/obj/item/cardpack/proc/zoom_out()
 	transform = matrix(0.4,0,0,0,0.4,0)
```
Comment (Rohesie) [inline code/game/objects/items/tcg/tcg.dm:154]:
```suggestion
/obj/item/cardpack/proc/zoom_out()
	transform = matrix(0.4,0,0,0,0.4,0)
	UnregisterSignal(src, list(COMISG_STORAGE_ENTERED, CONSIG_STORAGE_EXITED))

```

Comment (LemonInTheDark) [reply code/game/objects/items/tcg/tcg.dm:154]: why unregister here?

**Comment by LemonInTheDark** [comment]
It's not my fault I swear

File: code/game/atoms_movable.dm (hunk 4 above)
Comment (LemonInTheDark) [reply code/game/atoms_movable.dm]: adressed

File: code/game/atoms_movable.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/game/atoms_movable.dm]: addressed

File: code/game/atoms_movable.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/game/atoms_movable.dm]: addressed

File: code/game/atoms_movable.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/game/atoms_movable.dm]: addressed

File: code/game/objects/items/tcg/tcg.dm (hunk 5 above)
Comment (Rohesie) [reply code/game/objects/items/tcg/tcg.dm:154]: It's not unregistering anywhere.

Comment (LemonInTheDark) [reply code/game/objects/items/tcg/tcg.dm:154]: why does it need to unregister? do I need to do it on Destroy() or something?

Comment (Rohesie) [reply code/game/objects/items/tcg/tcg.dm:154]: Guess I misread it.

**Pull request merged**